homeassistant/components/mill/climate.py

```diff
@@ -111,26 +115,37 @@ def __init__(
         super().__init__(coordinator, device)
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
-        """Set new target temperature."""
+        """Set new target temperature and optionally HVAC mode."""
         if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
             return
         await self.coordinator.mill_data_connection.set_heater_temp(
             self._id, float(temperature)
         )
+        if (hvac_mode := kwargs.get(ATTR_HVAC_MODE)) is not None:
+            if hvac_mode in self._attr_hvac_modes:
```

Comment:
Checking if the mode is supported is already handled in the base climate entity.

Reply:
? Where? In https://github.com/home-assistant/core/blob/02dca5f0ad298a52818b6559ed17ce3c0ed075b6/homeassistant/components/climate/__init__.py#L759-L828 I only see temperature attributes being checked, and HVAC isn't part of the checking, as it just gets passed through via `kwargs[value] = temp`.

So that's why I'm adding this check here, because else it doesn't get checked.

Do I need to add this to the base service call, or can that be spun off into another bug?

Reply:
We can use this method:

https://github.com/home-assistant/core/blob/ab45460069b4110da9478d360c5162927578ccb1/homeassistant/components/climate/__init__.py#L573-L577

Reply:
That's not my question, this piece of code is on the receiving end of the service call, while that bit of code on homeassistant's side (the sending side) does not have validation of the HVAC mode when called with set_temperature

Do you want me to file a bug report on that, or include the fix in this PR?